docs(proto): fixed default serviceAddress and upgrade docs (#5236)
kumahq/kuma: UPGRADE.md

* The `/versions` endpoint was removed. This is not something that was reliable enough and version compatibility
is checked inside the DP
* We are deprecating `kuma.io/builtindns` and `kuma.io/builtindnsport` annotations in favour of the clearer `kuma.io/builtin-dns` and `kuma.io/builtin-dns-port`. The behavior of the new annotations is unchanged but you should migrate (a warning is present on the log if you are using the deprecated version).

> * By default, applications binding to `localhost` are not reachable anymore. We are changing the default inbound cluster from `localhost` to `inbound.address`. Before upgrade check if your applications are listening on `localhost` and should be exposed:

Reviewer: > We are changing the default inbound cluster from `localhost` to `inbound.address`

What is a "default inbound cluster"? This sentence doesn't make any sense to me.

Author: True, does `default service address` makes sense?

> * Applications that are binding to `localhost` won't be reachable anymore. We are changing the default inbound cluster that was always pointing to `localhost` to `DataplaneIP`. Before upgrade check if your applications are listening on `localhost` and should be exposed:

Reviewer 2: ```suggestion
* By default, applications binding to `localhost` are not reachable anymore. We are changing the default inbound cluster from `localhost` to `DataplaneIP`. Before upgrade check if your applications are listening on `localhost`:
```

>   * Universal: If you don't want to expose application make sure your application binds  to `127.0.0.1` and set `dataplane.networking.inbound[].serviceAddress: "127.0.0.1"`.

Reviewer: ```suggestion
  * Universal: If you don't want to expose an application make sure it binds to `127.0.0.1` and set `dataplane.networking.inbound[].serviceAddress: "127.0.0.1"`.
```

>   * Universal: If you don't want to expose application change binding to `127.0.0.1` and define `dataplane.networking.inbound[].serviceAddress: "127.0.0.1"`.

Reviewer 2: ```suggestion
  * Universal: If you don't want to expose application make sure your application binds  to `127.0.0.1` and set `dataplane.networking.inbound[].serviceAddress: "127.0.0.1"`.
```

>   * Kubernetes: Make sure your application binds to `0.0.0.0`.

Reviewer: Why do I have to do this? This line confuses me. Isn't it the same as for universal?

Author: On Universal, it's possible to change the serviceAddress. Charly mention to create separate sections for K8s and Universal

Reviewer: Why do I have to make sure the application is listening on `0.0.0.0`?

It feels like the docs should explain 4 cases:

My application is listening on `localhost` and:
* am on K8s and want it exposed
* am on K8s and don't want it exposed
* am on universal and want it exposed
* am on universal and don't want it exposed

> To ease migration you can temporarily disable this new behavior by setting `KUMA_DEFAULTS_ENABLE_LOCALHOST_INBOUND_CLUSTERS=true` on `kuma-cp`, this option will be removed in a future version.

Reviewer: ```suggestion
To make migration easier you can temporarily disable this new behavior by setting `KUMA_DEFAULTS_ENABLE_LOCALHOST_INBOUND_CLUSTERS=true` on `kuma-cp`, this option will be removed in a future version.
```

> Another way is to disable the new behavior by setting `kuma-cp` configuration `KUMA_DEFAULTS_ENABLE_LOCALHOST_INBOUND_CLUSTERS` to `true` or `defaults.enableLocalhostInboundClusters` to `true`. The last option is going to be removed in further versions.

Reviewer 2: ```suggestion
To ease migration you can temporarily disable this new behavior by setting `KUMA_DEFAULTS_ENABLE_LOCALHOST_INBOUND_CLUSTERS=true` on `kuma-cp`, this option will be removed in a future version.
```

* The `/versions` endpoint was removed. This is not something that was reliable enough and version compatibility
is checked inside the DP
* We are deprecating `kuma.io/builtindns` and `kuma.io/builtindnsport` annotations in favour of the clearer `kuma.io/builtin-dns` and `kuma.io/builtin-dns-port`. The behavior of the new annotations is unchanged but you should migrate (a warning is present on the log if you are using the deprecated version).

> * By default, applications binding to `localhost` are not reachable anymore. A `Dataplane` inbound's default `serviceAddress` is now the inbound's `address`. Before upgrade check if your applications are listening on `localhost` and:
>   * you run on K8s and you want it exposed - change binding of the application to `0.0.0.0`.
>   * you run on K8s and you don't want it exposed - changes are not needed.
>   * you run on Universal and you want it exposed - change it binding to `inbound.address` or set `dataplane.networking.inbound[].serviceAddress: "127.0.0.1"`.
>   * you run on Universal and you don't want it exposed - changes are not needed.
> To make migration easier you can temporarily disable this new behavior by setting `KUMA_DEFAULTS_ENABLE_LOCALHOST_INBOUND_CLUSTERS=true` on `kuma-cp`, this option will be removed in a future version.

Reviewer: ```suggestion
* By default, applications binding to `localhost` are not reachable anymore. A `Dataplane` inbound's default `serviceAddress` is now the inbound's `address`. Before upgrade, if you have applications listening on `localhost` that you want to expose on:
  * Kubernetes: listen on `0.0.0.0` instead
  * Universal: listen on `inbound.address` instead or set `dataplane.networking.inbound[].serviceAddress: "127.0.0.1"`
  To make migration easier you can temporarily disable this new behavior by setting `KUMA_DEFAULTS_ENABLE_LOCALHOST_INBOUND_CLUSTERS=true` on `kuma-cp`, this option will be removed in a future version.
```
